src/validation/rules/UniqueInputFieldNamesRule.ts

```diff
@@ -34,18 +37,54 @@ export function UniqueInputFieldNamesRule(
         knownNames = prevKnownNames;
       },
     },
+    ListValue: {
+      enter(node) {
+        node.values.forEach((valueNode) => {
+          if(valueNode.kind === 'ObjectValue') {
+            knownNamesInList.push(valueNode.fields);
+          }
+        });
+      },
+      leave() {
+        knownNamesInList = Object.create([]);
```

Comment:
I think this would also need to use a stack else it would make list --> object --> field --> list lose track of the outer object when it bubbles back up

Reply:
Let me address the case of this `list --> object --> field --> list`.

Reply:
I’ve changed the approach to use a stack to hold `listValue[]` and pop the state when it exits. I’ve also added a test for this case. Could you please help me review it again? @JoviDeCroock